cabal-install/src/Distribution/Client/IndexUtils.hs

```diff
@@ -754,38 +753,26 @@ lazyUnfold step = goLazy . Just
       vs' <- goLazy mk'
       return ((k, v) : vs')
 
--- | Which index do we mean?
-data Index
-  = -- | The main index for the specified repository
-    RepoIndex RepoContext Repo
```

Comment:
I'm unsure about this. Maybe it's better to keep `Repo` and `RepoCtx` together since they are closely related